scripts/image-build/build-vyos-image

```diff
@@ -644,3 +652,8 @@ Pin-Priority: 600
                 target = f"{os.path.splitext(raw_image)[0]}.{image_ext}"
                 print(f"I: Building {f} file {target}")
                 cmd(f"qemu-img convert -f raw -O {f} {image_opts} {raw_image} {target}")
+
+        manifest['artifacts'].append(raw_image)
```

Comment:
This will not us allow to use the manifest to locate qcow2, compressed raw, OVA, Hyper-V .vhdx, etc.

The main problem, of course, is that names of those artifacts aren't present anywhere at the moment... I wonder if we should add a new flavor field like `artifact_extensions` so that the script can look for files with those extensions and include them in the manifest, or raise an error if an apparently successful build didn't in fact produce files with those extensions.

Reply:
Extending the `build_config` with a new key `artifact_extensions` sounds reasonable, thus we can autodetect the files when generating the manifest file at the end.

Could you add this to the build script @dmbaturin ?

Reply:
If you don't mind taking it up, I think it's ok to extend the scope of this PR to also add that option, since the goal cannot be accomplished without it.

I'm actually wondering if the build script itself should use that information to copy just the artifacts to, say, `build/artifacts` so that the manifest is purely informational and if you want artifacts, you can just copy everything from that dir without even looking into the manifest.

Reply:
PR updated to address todays meetings resolution